### PR TITLE
Fix events for instance create/update

### PR DESCRIPTION
**Pull request by barthy1**
- only one instance action is shown
- when instance is recreated the action is `recreate` instead of start or update
- az is shown for instance creation case

[#115793643](https://www.pivotaltracker.com/story/show/115793643)

Signed-off-by: Konstantin Maksimov <kmaksimov@ru.ibm.com>

### CODE REVIEW

File: spec/shared/support/table_helpers.rb [hunk 1]
```diff
@@ -28,7 +28,7 @@ def clean(content)
       end
 
       def parse(content)
-        CSV.parse(clean(content), { col_sep: "|" }).map(&:compact)
+        CSV.parse(clean(content), { col_sep: "|", quote_char: "\x00" }).map(&:compact)
```
Comment (dpb587-pivotal) [inline spec/shared/support/table_helpers.rb]: Why is this change needed? Tests seem to run fine without it.

File: spec/integration/template_evaluation_spec.rb [hunk 2]
```diff
@@ -55,6 +55,15 @@
...
+      {'ID' => /[0-9]{1,3} <- [0-9]{1,3}/, 'Time' => 'xxx xxx xx xx:xx:xx UTC xxxx', 'User' => 'test', 'Action' => 'update', 'Object type' => 'deployment', 'Task' => /[0-9]{1,3}/, 'Object ID' => 'simple', 'Dep' => 'simple', 'Inst' => '-', 'Context' => '-'},
+      {'ID' => /[0-9]{1,3} <- [0-9]{1,3}/, 'Time' => 'xxx xxx xx xx:xx:xx UTC xxxx', 'User' => 'test', 'Action' => 'recreate', 'Object type' => 'instance', 'Task' => /[0-9]{1,3}/, 'Object ID' => 'id_job/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx', 'Dep' => 'simple', 'Inst' => 'id_job/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx', 'Context' => '-'},
+      {'ID' => /[0-9]{1,3}/, 'Time' => 'xxx xxx xx xx:xx:xx UTC xxxx', 'User' => 'test', 'Action' => 'recreate', 'Object type' => 'instance', 'Task' => /[0-9]{1,3}/, 'Object ID' => 'id_job/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx', 'Dep' => 'simple', 'Inst' => 'id_job/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx', 'Context' => '-'},
+      {'ID' => /[0-9]{1,3}/, 'Time' => 'xxx xxx xx xx:xx:xx UTC xxxx', 'User' => 'test', 'Action' => 'update', 'Object type' => 'deployment', 'Task' => /[0-9]{1,3}/, 'Object ID' => 'simple', 'Dep' => 'simple', 'Inst' => '-', 'Context' => '-'},
+    )
```
Comment (dpb587-pivotal) [inline spec/integration/template_evaluation_spec.rb]: When running this test, the events output includes the following seemingly-duplicate lines - two each for the `create` and `delete` event. It seems separate from the changes here, but curious if it's expected.

```
| 13 <- 12 | Wed Apr 27 00:03:12 UTC 2016 | test | create   | vm           | 13540                                       | 5    | simple | id_job/c50eeb84-fa56-40be-8755-33faa968b881 | -       |
| 12       | Wed Apr 27 00:03:12 UTC 2016 | test | create   | vm           | -                                           | 5    | simple | id_job/c50eeb84-fa56-40be-8755-33faa968b881 | -       |
| 11 <- 10 | Wed Apr 27 00:03:12 UTC 2016 | test | delete   | vm           | 13349                                       | 5    | simple | id_job/c50eeb84-fa56-40be-8755-33faa968b881 | -       |
| 10       | Wed Apr 27 00:03:12 UTC 2016 | test | delete   | vm           | 13349                                       | 5    | simple | id_job/c50eeb84-fa56-40be-8755-33faa968b881 | -       |
```

File: spec/shared/support/table_helpers.rb (hunk 1 above)
Comment (dpb587-pivotal) [inline spec/shared/support/table_helpers.rb]: Nevermind - I see it's used by #1221.

File: spec/integration/template_evaluation_spec.rb (hunk 2 above)
Comment (barthy1) [inline spec/integration/template_evaluation_spec.rb]: @dpb587-pivotal: yes, this behaviour is expected. In case of instance recreation, current vm should be deleted (10, 11) and new vm should be created (12, 13). vm creation/deletion events are ending actions (https://github.com/cloudfoundry/bosh-notes/blob/master/events.md), so each event has starting and ending records. It's not important for "deploy --recreate" case, so we didn't add these lines to the table check.
